### PR TITLE
Rewrite and improve documentation

### PR DESCRIPTION
**Pull request by elliotdavies**
- Revamp README with better introduction to library, etc
- Add documentation and inline examples for all exported functions

Fixes https://github.com/times/data-validator/issues/15.

For a future piece of work I think it'd be good to add an `examples` folder or similar to show more complex use-cases such as nested object schemas.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,141 +1,678 @@
...
+
+isErr(validate('thirty')); //=> true
+```
+
+#### validateIsIn
+
+Validates that the data is in the specified array, according to the ES6 [`.includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) method.
+
+```js
+const validate = validateIsIn(['a', 'b']);
```
Comment (chrishutchinson) [inline README.md]: Out of curiosity, if I did the following, would it work, or be an error?

```js
const validate = validateIsIn('foobar');

validate('foo'); //=> ????
```

This is given that othings other than arrays support `.includes`

File: README.md [hunk 2]
```diff
@@ -1,141 +1,678 @@
...
+  validateStartsWithA,
+]);
+
+getErrors(validateWord(123)); //=> [`123 failed to typecheck (expected string)`]
+
+getErrors(validateWord('abracadabra')); //=> [`"abracadabra" was too long`]
+
+getErrors(validateWord('tea')); //=> [`"tea" did not start with A`]
+
+isOK(validateWord('Andy')); //=> true
```
Comment (chrishutchinson) [inline README.md]: ☹️

File: README.md [hunk 3]
```diff
@@ -1,141 +1,678 @@
...
+const results = [err('Error one'), ok(), err(['Error two', 'Error three'])];
+
+const combinedResult = concatResults(results);
+
+getErrors(mergedResult); //=> [`Error one`, `Error two`, `Error three`]
+```
+
+### Extracting errors
+
+To extract errors from a `Result` you should normally use the `getErrors` function. In future there will be several functions available that will format the errors in different ways.
```
Comment (chrishutchinson) [inline README.md]: >In the future, there will be various input formats, but for now, we'll just copy and paste

**Review by chrishutchinson — COMMENTED** [review]
Looking great! Just one main question, which is mostly just curiosity

File: README.md (hunk 1 above)
Comment (elliotdavies) [reply README.md]: I think that'd work, yep

**Pull request merged**